### DISCUSSION TITLE
Extracting list of barcodes in a UMAP cluster

### DISCUSSION THREAD

**Discussion by zrtehrani** [satijalab/seurat#6908] [Q&A]
As a beginner in bioinformatics, I've started an analysis using Seurat. My cells were clustered using UMAP, and now I need to further analyze a specific cluster. There are two questions I would like to ask:

1-	How can I have a list of all the cells in a specific UMAP cluster?
2-	How can I have the feature barcode matrix for a specific UMAP cluster?

Thanks,
Z

**Comment by Dragonmasterx87** [comment]
Hi @zrtehrani 

There are a couple of ways to do this. The easiest way is to use `CellsByIdentities`

Say your seurat object is called `cowabunga` containing a metadata slot called `cast` containing 6 types of metadata: `april`, `splinter`, `raphael`, `michelangelo`, `donatello`, `leonardo`, `shredder`, `beebop` and `krang`.

```
# I want all the cell names of all cells falling under michelangelo:
Idents(cowabunga) <- 'cast'
goodguys <- CellsByIdentities(cowabunga, idents = 'michelangelo')

# Good guys should be a dataframe containing your cell identities corresponding to the cluster micheangelo
```

You can read more about this [here](https://mojaveazure.github.io/seurat-object/reference/CellsByIdentities.html#examples).

If you have more questions ask me by tagging or I wont know you have a question.

Cheers,

🐲
